include/bl_monotonic_counters.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BL_MONOTONIC_COUNTERS_H_
+#define BL_MONOTONIC_COUNTERS_H_
+
+#include <string.h>
+#include <zephyr/types.h>
+#include <drivers/nrfx_common.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** A single monotonic counter. It consists of a description value, a 'type',
+ *  to know what this counter counts. Further, it has a number of slots
+ *  allocated to it. Each time the counter is updated, a new slot is written.
+ *  This way, the counter can be updated without erasing anything. This is
+ *  necessary so the counter can be used in the OTP section of the UICR
+ *  (available on e.g. nRF91 and nRF53).
+ */
+struct monotonic_counter {
+	uint16_t description; /* Counter description. What the counter is used for. See COUNTER_DESC_*. */
+	uint16_t num_counter_slots; /* Number of entries in 'counter_slots' list. */
+	uint16_t counter_slots[1];
+};
+
+/** The second data structure in the provision page. It has unknown length since
+ *  'counters' is repeated. Note that each entry in counters also has unknown
+ *  length, and each entry can have different length from the others, so the
+ *  entries beyond the first cannot be accessed via array indices.
+ */
```

Comment:
```suggestion
/** This struct is placed after the bl_storage_data struct if present. It has unknown length since
 *  'counters' is repeated. Note that each entry in counters also has unknown
 *  length, and each entry can have different length from the others, so the
 *  entries beyond the first cannot be accessed via array indices.
 */
```